### PR TITLE
Store onboarding state in user metadata

### PR DESCRIPTION
**Pull request by poulch**
To restart onboarding state you have to remove metadata, using grapiQL:
1.  Go to you_env_url/graphql 
2. Put this mutation:
```
mutation clearMeMetadata($id: ID!, $keys: [String!]!) {
  deleteMetadata(id: $id, keys: $keys) {
    errors {
      field
      message
    }
  }
}
```
3. Add variables:
```
{
  "id": YOUR_USER_ID,
  "keys": ["onboarding"]
}
```
4. Add a headers:
```
{
   authorization-bearer: YOUR_ACCESS_TOKEN
}
```

You can take your id from this query:
```
query GetID {
  me {
    id
  }
}
```

## What type of PR is this?
- [ ] 💅 Refactor
- [ ] 🌟 Feature
- [ ] 🔥 Bug Fix
- [ ] 🔩 Maintenance
- [ ] 🛠 Workflow CI/CD changes

## Related Issues or Documents

- closes #

## Have you written tests?
- [ ] Yes!
- [ ] No... here is why: _Writing tests are mandatory, please replace this text with why test are not included in this PR_

### CODE REVIEW

File: src/newHome/homeOnboarding/onboardingContext/useOnboardingStorage.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        return;
+      }
+
+      try {
+        const userMetadata: Array<MetadataInput> = [
+          ...(user?.metadata?.map(data => ({
+            key: data.key,
+            value: data.value,
+          })) ?? []),
+        ];
```
Comment (Cloud11PL) [inline src/newHome/homeOnboarding/onboardingContext/useOnboardingStorage.ts:45]:
```suggestion
const userMetadata: MetadataInput[] = user?.metadata 
  ? user.metadata.map(data => ({
      key: data.key,
      value: data.value
    }))
  : [];
```

File: src/newHome/homeOnboarding/onboardingContext/useOnboardingStorage.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        } else {
+          userMetadata.push({
+            key: METADATA_KEY,
+            value: metadataValue,
+          });
+        }
+
+        await updateMetadata({
+          variables: {
+            id: user?.id ?? "",
```
Comment (Cloud11PL) [inline src/newHome/homeOnboarding/onboardingContext/useOnboardingStorage.ts:63]: Suggestion: We're already returning early if user is not present so `user.id` should be fine here I think

**Review by Cloud11PL — DISMISSED** [review]

**Review by Cloud11PL — APPROVED** [review]

File: src/newHome/mutations.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import { gql } from "@apollo/client";
+
+export const updateUserMetadata = gql`
+  mutation UpdateUserMetadata($id: ID!, $input: [MetadataInput!]!) {
```
Comment (andrzejewsky) [inline src/newHome/mutations.ts:4]: this is used only for onboarding so you can call it eg "SaveOnBoardingState", update metadata sounds like generic solution that can be used anywhere

File: src/newHome/homeOnboarding/onboardingContext/useOnboardingStorage.ts [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+          userMetadata[metadataIndex] = {
+            key: METADATA_KEY,
+            value: metadataValue,
+          };
+        } else {
+          userMetadata.push({
+            key: METADATA_KEY,
+            value: metadataValue,
+          });
+        }
```
Comment (andrzejewsky) [inline src/newHome/homeOnboarding/onboardingContext/useOnboardingStorage.ts:58]: preparing metadata to be saved can be a separated function

File: src/newHome/homeOnboarding/onboardingContext/useOnboardingStorage.ts [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+      if (!metadata) {
+        return undefined;
+      }
+
+      return JSON.parse(metadata.value);
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.warn("Could not get onboarding state from metadata", { error });
+
+      return undefined;
```
Comment (andrzejewsky) [inline src/newHome/homeOnboarding/onboardingContext/useOnboardingStorage.ts:29]: what happens if this occur? do you have some default state?

**Review by andrzejewsky — APPROVED** [review]

File: src/newHome/homeOnboarding/onboardingContext/useOnboardingStorage.ts (hunk 5 above)
Comment (poulch) [reply src/newHome/homeOnboarding/onboardingContext/useOnboardingStorage.ts:29]: nothing happened as state is kept in react state hook so the user refresh the page state will be restarted to last one saved in metadata or initial one when never was saved in metadata

**Pull request merged**